enginetest/enginetests.go

```diff
@@ -2944,6 +2944,13 @@ func TestChecksOnInsert(t *testing.T, harness Harness) {
 
 	AssertErr(t, e, harness, "INSERT INTO t1 (a,b) select a - 2, b - 1 from t2", sql.ErrCheckConstraintViolated)
 	RunQuery(t, e, harness, "INSERT INTO t1 (a,b) select a, b from t2")
+
+	// Check that INSERT IGNORE correctly drops errors with check constraints and does not update the actual table.
+	RunQuery(t, e, harness, "INSERT IGNORE INTO t1 VALUES (5,2, 'abc')")
```

Comment:
Good test case, but have one where you insert two rows in one statement, and only one violates the check. The other row should still be inserted.

Reply:
I don't think that is correct mysql behavior. If one value in an insert statement fails, then mysql does not accept any value

Reply:
Nope:

```
mysql> show create table checks;
+--------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table  | Create Table
                                                                                     |
+--------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| checks | CREATE TABLE `checks` (
  `a` int NOT NULL,
  `b` int DEFAULT NULL,
  PRIMARY KEY (`a`),
  CONSTRAINT `checks_chk_1` CHECK ((`b` > 10))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+--------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+


mysql> insert ignore into checks values (1,1), (2,11);
Query OK, 1 row affected, 1 warning (0.01 sec)
Records: 1  Duplicates: 0  Warnings: 1

mysql> select * from checks;
+---+------+
| a | b    |
+---+------+
| 2 |   11 |
+---+------+
1 row in set (0.01 sec)
```

Reply:
My fault assumed vanilla insert. You are correct